doc/sphinx/src/using.rst

```diff
@@ -194,6 +194,73 @@ exceptions. For very large negative inputs, it returns 0. For very
 large positive ones, it returns
 ``std::numeric_limits<T>::infinity()``.
 
+The function
+
+.. code-block:: cpp
+
+  template <typename DataType>
+  PORTABLE_FUNCTION constexpr DataType reldiff(const DataType actual, const DataType expected)
+
+returns the relative difference between the expected value and the actual
+result (:math:`\left|(A-E)/E\right|`).  If the expected and actual values are
+both zero, this is defined to be a relative difference of 0%.  If the expected
+value is zero and the actual value is non-zero, this is defined to be a
+relative difference of 100%.  Unlike ``ratio``, ``reldiff`` is tuned to get
+meaningful results for the use-case of unit tests, rather than for performance.
```

Comment:
As we discussed on MM, I found this an odd model, given we are working with floating point numbers. If this is a context where exact 0 can be expected, we should document that. Otherwise we might want to switch to something that switches to an absolute error if expected is exactly 0.

Either ways is fine, so long as this is named/documented to be what you need for singe. If we end up going with the latter case, then I can use this instead of the in-house function in singularity-eos.

Reply:
I went back and looked at the usage of `reldiff` in Singe (which is where it came from).  It's pretty much exclusively (other than one or two printouts) used in the form `if(reldiff(actual, expected) <= tolerance)`.  In that case, we can trivially rewrite those clauses to be `if(abs(actual-expected) <= tolerance * abs(expected))`, which eliminates the entire divide-by-zero question.  Why I didn't do that earlier is unclear.

So what might be more useful than deciding on the "best" way to write `reldiff`, maybe we should just drop it from here and at some point I may change Singe to the other form.  But in that case, it _would_ be useful to add a GPU-compatible `abs` to `math_utils.hpp`.  What do you think about that plan?

Reply:
PS: I should say "eliminates the entire divide-by-zero question _and_ makes it explicit and under the user's control what to do if expected is zero".

I'm also hesitant to reproduce something like the Catch2 matchers for GPU, given my relatively simple use-cases.

Reply:
That's fine by me. We could alternatively implement something like the check you actually use in singe, sans the divide by zero.

Reply:
After our discussion, let's just punt on this and remove reldiff as you suggest. :+1: